src/api.ts

```diff
@@ -398,22 +403,33 @@ export namespace conversion {
     });
   }
 
-  export async function openapi(document: vscode.Uri | string): Promise<any> {
-    let endpoint = `${apiRootPath}/${resource}/convert?type=openapi`;
+  export async function openapi(openAPIDocument: vscode.Uri | string, openAPIConfig?: SharedNs.WebviewCommandPayloadOpenAPISelectRequests, add?: vscode.Uri | string): Promise<any> {
+   
+    let endpoint = `${apiRootPath}/adapterDefinitions/convert?type=openapi`;
     log.debug(`Calling '${endpoint}'`);
     let client = await getClient();
     const form = new FormData();
     // part 1
-    form.append('sourceFile', document instanceof vscode.Uri ? fs.readFileSync(document.fsPath, 'utf8') : document);
+    form.append('sourceFile', openAPIDocument instanceof vscode.Uri ? fs.readFileSync(openAPIDocument.fsPath, 'utf8') : openAPIDocument);
+    log.debug("Set 'sourceFile'");
+    
+    // part 2
+    if (add) {
+      log.debug("Set 'targetADD'");
+      form.append('targetADD', add instanceof vscode.Uri ? fs.readFileSync(add.fsPath, 'utf8') : add);
+    }
 
-    try {
-      let res = await callAPI(() => client.post(endpoint, form) as Promise<AxiosResponse<any>>);
-      return res.data;
-    } catch (err) {
+    if (openAPIConfig) {
+      log.debug("Set 'openAPIConverterConfigRequest'", log.format(openAPIConfig));
+      form.append('openAPIConverterConfigRequest', JSON.stringify(openAPIConfig));
+    }
+
+    return callAPI(() => client.post(endpoint, form) as Promise<AxiosResponse<OpenAPINS.Root>>, (err) => {
       if (err instanceof AxiosError && err.response?.status !== 404) {
-        log.infoServer(err.response?.data);
+        logInfoServer(err.response?.data);
       }
-      throw new RABError(`Failed to call 'POST ${endpoint}'`, err);
-    }
+      showErrorMessage("❌ Conversion failed");
```

Comment:
Let's try to decouple api layer from UI interaction. Throwing the error out and let the upper layer decide how to handle it.

Reply:
The error is shown in another place. We didn't hide it.